sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -274,115 +273,229 @@ export abstract class ECertFileHandler extends ESDCFileHandler {
+      await this.processFile(
+        fileProcessingSummary,
+        eCertIntegrationService,
+        filePath,
+        eCertFeedbackErrorCodeMap,
       );
     }
-    return processFiles;
   }
 
   /**
    * Process each individual E-Cert response file from the SFTP.
+   * @param processSummary process summary of file processing.
    * @param eCertIntegrationService
    * @param filePath E-Cert response file to be processed.
-   * @returns Process summary and errors summary.
+   * @param offeringIntensity offering intensity.
+   * @param eCertFeedbackErrorCodeMap e-Cert feedback error map
+   * to get error id by error code.
    */
   private async processFile(
+    processSummary: ProcessSummary,
     eCertIntegrationService: ECertIntegrationService,
     filePath: string,
-  ): Promise<ProcessSFTPResponseResult> {
-    const result = new ProcessSFTPResponseResult();
-    result.processSummary.push(`Processing file ${filePath}.`);
-
-    let responseFile: ECertFullTimeResponseRecord[];
-
+    eCertFeedbackErrorCodeMap: Record<string, number>,
+  ): Promise<void> {
+    processSummary.info(`Processing file ${filePath}.`);
+    let eCertFeedbackResponseRecords: ECertResponseRecord[];
     try {
-      responseFile = await eCertIntegrationService.downloadResponseFile(
-        filePath,
-      );
+      eCertFeedbackResponseRecords =
+        await eCertIntegrationService.downloadResponseFile(filePath);
     } catch (error) {
-      this.logger.error(error);
-      result.errorsSummary.push(`Error downloading file ${filePath}. ${error}`);
       // Abort the process nicely not throwing an exception and
       // allowing other response files to be processed.
-      return result;
+      processSummary.error(`Error downloading file ${filePath}.`, error);
+      return;
     }
-
-    result.processSummary.push(`File contains ${responseFile.length} records.`);
-
-    for (const feedbackRecord of responseFile) {
-      try {
-        await this.processErrorCodeRecords(feedbackRecord);
-        this.logger.log(
-          `Successfully processed line ${feedbackRecord.lineNumber}.`,
+    // Processing the records.
+    processSummary.info(
+      `File contains ${eCertFeedbackResponseRecords.length} records.`,
+    );
+    try {
+      this.sanitizeErrorCodes(
+        eCertFeedbackResponseRecords,
+        eCertFeedbackErrorCodeMap,
+      );
+      for (const eCertFeedbackResponseRecord of eCertFeedbackResponseRecords) {
+        const recordProcessSummary = new ProcessSummary();
+        processSummary.children(recordProcessSummary);
+        await this.createDisbursementFeedbackError(
+          recordProcessSummary,
+          eCertFeedbackResponseRecord,
+          eCertFeedbackErrorCodeMap,
+        );
+      }
+    } catch (error: unknown) {
+      // Any error caught here will abort the file processing.
+      processSummary.error(`Error processing the file ${filePath}.`, error);
+      return;
```

Comment:
Does the return add any value here?

Reply:
I just realized it is the end of the method 😊. The answer is no and I will remove it.